pkg/driver/mount.go

```diff
@@ -271,7 +255,20 @@ func (m *S3Mounter) Mount(bucketName string, target string,
 
 	env := []string{}
 	if credentials != nil {
-		env = credentials.Env()
+		var awsProfile AWSProfile
```

Comment:
I think that would cause writing credentials to a file even though there are no mounts at all. I prefer keeping these files specific to mounts, in future we might support passing long-term credentials with `authenticationSource: pod`, allow running Mountpoint processes as different users etc. and this approach would play with them nicely.